src/probability/variance.lean

```diff
@@ -30,28 +39,166 @@ open_locale big_operators measure_theory probability_theory ennreal nnreal
 
 namespace probability_theory
 
-/-- The variance of a random variable is `𝔼[X^2] - 𝔼[X]^2` or, equivalently, `𝔼[(X - 𝔼[X])^2]`. We
-use the latter as the definition, to ensure better behavior even in garbage situations. -/
+/-- The `ℝ≥0∞`-valued variance of a real-valued random variable defined as a lintegral.-/
+def evariance {Ω : Type*} {m : measurable_space Ω} (f : Ω → ℝ) (μ : measure Ω) : ℝ≥0∞ :=
+∫⁻ ω, ∥f ω - μ[f]∥₊^2 ∂μ
+
+/-- The `ℝ`-valued variance of a real-valued random variable defined by applying `ennreal.to_real`
+to `evariance`. -/
 def variance {Ω : Type*} {m : measurable_space Ω} (f : Ω → ℝ) (μ : measure Ω) : ℝ :=
-μ[(f - (λ ω, μ[f])) ^ 2]
+(evariance f μ).to_real
+
+lemma _root_.measure_theory.mem_ℒp.evariance_lt_top {Ω : Type*} {m : measurable_space Ω}
+  {f : Ω → ℝ} {μ : measure Ω} [is_finite_measure μ] (hf : mem_ℒp f 2 μ) :
```

Comment:
Could you add a `variables` declaration for Omega, m, f, µ, and remove them from the lemmas?